pkg/tbtc/wallet.go

```diff
@@ -617,11 +624,40 @@ func EnsureWalletSyncedBetweenChains(
 			}
 
 			if isDeposit {
-				// If that's the case, the wallet was already done their
-				// first Bitcoin transaction and the Bridge is awaiting the
-				// SPV proof.
+				// If that's the case, the wallet has already created a deposit
+				// sweep as their first Bitcoin transaction and the Bridge is
+				// awaiting the SPV proof.
+				return fmt.Errorf("wallet already produced their first " +
+					"Bitcoin transaction (deposit sweep); Bridge is probably " +
+					"awaiting the SPV proof",
+				)
+			}
+
+			// The transaction could be a moved funds sweep request. In that
+			// case the transaction's input must refer to a moved funds sweep
+			// request. If the input points to a moved funds sweep request, that
+			// means the given transaction is produced by our wallet.
+			_, isRequest, err := bridgeChain.GetMovedFundsSweepRequest(
+				input.Outpoint.TransactionHash,
+				input.Outpoint.OutputIndex,
+			)
+			if err != nil {
+				return fmt.Errorf(
+					"cannot get moved funds sweep request for hash [%s] "+
+						"and output index [%v]: [%v]",
+					input.Outpoint.TransactionHash.String(),
+					input.Outpoint.OutputIndex,
+					err,
+				)
+			}
+
+			if isRequest {
+				// If that's the case, the wallet has already created a moved
+				// funds sweep their first Bitcoin transaction and the Bridge
```

Comment:
```suggestion
				// funds sweep as their first Bitcoin transaction and the Bridge
```

Reply:
Done in c5768c558ccac5d71684a8be05eff7abf489e26b.